piptools/resolver.py

```diff
@@ -100,34 +109,36 @@ def combine_install_requirements(repository, ireqs):
 class Resolver:
     def __init__(
         self,
-        constraints,
-        repository,
-        cache,
-        prereleases=False,
-        clear_caches=False,
-        allow_unsafe=False,
-    ):
+        constraints: List[InstallRequirement],
+        repository: BaseRepository,
+        cache: DependencyCache,
+        prereleases: Optional[bool] = False,
```

Comment:
Can we drop Optional?

```suggestion
        prereleases: bool = False,
```